pycobertura/reporters.py

```diff
@@ -249,61 +221,23 @@ def __init__(self, *args, **kwargs):
         self.show_missing = kwargs.pop("show_missing", True)
         super(HtmlReporterDelta, self).__init__(*args, **kwargs)
 
-    def get_source_hunks(self, filename):
-        hunks = self.differ.file_source_hunks(filename)
-        return hunks
-
-    def format_row(self, row):
-        total_statements = "%+d" % row.total_statements if row.total_statements else "-"
-        total_misses = "%+d" % row.total_misses if row.total_misses else "-"
-        line_rate = "%+.2f%%" % (row.line_rate * 100) if row.line_rate else "-"
-
-        if self.show_source is True and self.show_missing is True:
-            missed_lines = [
-                "%s%d" % (["-", "+"][is_new], lno) for lno, is_new in row.missed_lines
-            ]
-
-        row_values = [
-            row.filename,
-            total_statements,
-            total_misses,
-            line_rate,
-        ]
-
-        if self.show_source is True and self.show_missing is True:
-            row_values.append(missed_lines)
-            row = file_row_missed(*row_values)
-        else:
-            row = file_row(*row_values)
-
-        return row
-
     def generate(self):
         lines = self.get_report_lines()
 
-        formatted_lines = []
-        for row in lines:
-            formatted_row = self.format_row(row)
-            formatted_lines.append(formatted_row)
-
-        if self.show_source is True:
-            sources = []
-            for filename in self.differ.files():
-                source_hunks = self.get_source_hunks(filename)
-                if not source_hunks:
-                    continue
-                sources.append((filename, source_hunks))
-
         template = env.get_template("html-delta.jinja2")
 
         render_kwargs = {
-            "lines": formatted_lines[:-1],
-            "footer": formatted_lines[-1],
+            "lines": lines[:-1],
+            "footer": lines[-1],
             "show_missing": self.show_missing,
             "show_source": self.show_source,
         }
 
         if self.show_source is True:
-            render_kwargs["sources"] = sources
+            render_kwargs["sources"] = [
+                (filename, self.differ.file_source_hunks(filename))
+                for filename in self.differ.files()
+                if self.differ.file_source_hunks(filename)
```

Comment:
This line is being computed twice, how can we avoid doing this?

Reply:
done